projects/angular/src/utils/list-key-navigation/key-navigation-list.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Directive, ElementRef, Inject, Input, Renderer2 } from '@angular/core';
+import {
+  focusElement,
+  getFlattenedFocusableItems,
+  initializeKeyListItems,
+  KeyNavigationCode,
+  setActiveKeyListItem,
+  validKeyNavigationCode,
+} from '@cds/core/internal';
+
+import { getNextKeyListItem } from './utils';
+
+export interface KeyNavigationListConfig {
+  keyListItems: string;
+  layout: 'both' | 'horizontal' | 'vertical';
+  manageFocus: boolean;
+  manageTabindex: boolean;
+  loop: boolean;
+  dir: string | null;
+}
+
+@Directive({
+  selector: '[clrKeyNavigationList]',
+})
+export class ClrKeyNavigationList {
+  private get listItems(): NodeListOf<HTMLElement> {
+    return this.host.nativeElement.querySelectorAll(
+      [
+        'a[href]',
+        'area[href]',
+        'input:not([disabled])',
+        'button:not([disabled])',
+        'select:not([disabled])',
+        'textarea:not([disabled])',
+        'iframe',
+        'object',
+        'embed',
+        '*[tabindex]', // -1 tabindex is a focusable element and needed for keyboard navigation
+        '*[contenteditable=true]',
+        '[role=button]:not([disabled])',
+      ].join(',')
+    );
+  }
+
+  private config: KeyNavigationListConfig;
+
+  constructor(
+    private host: ElementRef<HTMLElement>,
+    private renderer: Renderer2,
+    @Inject('ClrListKeyNavigation') config: Partial<KeyNavigationListConfig>
+  ) {
+    this.config = {
+      keyListItems: 'keyListItems',
+      layout: 'vertical',
+      manageFocus: true,
+      manageTabindex: true,
+      loop: false,
+      dir: this.host.nativeElement.getAttribute('rtl'),
+      ...config,
+    };
+  }
+
+  @Input()
+  set configuration(config: Partial<KeyNavigationListConfig>) {
+    this.config = {
+      ...this.config,
```

Comment:
I think the default config should be used here so that behavior is more consistent/predictable if this setter is called more than once.

Reply:
```typescript
clrKeyNavigationList.configuration = { layout: 'horizontal' };

// Due to the normal nature of assignment, I think a user would assume that this assignment
// does the same thing whether or not there was some previous assignment.
clrKeyNavigationList.configuration = { manageFocus: false };
```